packages/@aws-cdk/pipelines/lib/lambda/index.py

```diff
@@ -0,0 +1,35 @@
+import boto3
+import json
+
+cp = boto3.client('codepipeline')
+
+def handler(event, context):
+  pipeline_name = event['PipelineName']
+  stage_name = event['StageName']
+  action_name = stage_name + 'ManualApproval'
+
+  response = cp.get_pipeline_state(name=pipeline_name)
+
+  token = parse_pipeline_state(response, stage_name, action_name)
+
+  if token:
```

Comment:
I think we might need to wait a reasonable time for the token to become available if it's not available right away.

Reply:
Yes! Haven't implemented that part yet! I was also thinking about running the ManualApproval and SecurityCheck in parallel.. only downside is that a customer might approve the changes **before** the SecurityCheck runs..

Reply:
Yeah exactly. And the passthrough of the console URL won't work that way. Better to do them in sequence and wait for a bit.

 Don't forget to configure `timeout` on the Lambda function.